src/java.base/share/classes/sun/nio/ch/NioSocketImpl.java

```diff
@@ -564,7 +564,7 @@ protected void connect(SocketAddress remote, int millis) throws IOException {
             throw new IOException("Unsupported address type");
         InetSocketAddress isa = (InetSocketAddress) remote;
         if (isa.isUnresolved()) {
-            throw new UnknownHostException(isa.getHostName());
+            throw new UnknownHostException("Unresolved address: " + isa.getHostName());
```

Comment:
The changes means this change isn't needed now. I'm really tempted to say we change the unsupported address type to throw UOE, and the unresolved case to throw IAE, but don't have the complicated this PR any more, plus the SocketImpl API is hugely under specified so might result in more discussion that is worth it.

Reply:
Reverted in d3568d16cd227940d9254bc4212f332f89243bc2.